django_emarsys/config.py

```diff
@@ -0,0 +1,160 @@
+    check.
+
+    The extra `hasattr()` check is necessary to make this work with overridden
+    settings from tests. If a setting existed in the original settings object,
+    but is then deleted in overridden settings, then `dir()` will report it
+    present, but accessing it raises an `AttributeError`.
+
+    Minimal test for this:
+
+    >>> from django.conf import settings
+    >>> from django.test.utils import override_settings
+    >>> @override_settings()
+    >>> def test_strange_dir_issue():
+    >>>     delattr(settings, 'DEBUG')
+    >>>     assert 'DEBUG' in dir(settings)
+    >>>     assert not hasattr(settings, 'DEBUG')
+    >>>
+    >>> test_strange_dir_issue()
+
+    """
+    return validate_configuration({key: getattr(settings, key)
+                                  for key in dir(settings)
+                                  if hasattr(settings, key)})
+
+
+def flatten(l):
+    """
+    Flatten a list of lists into a list
+    """
+    return list(chain.from_iterable(l))
+
+
+def duplicates(l):
+    """
+    Return non-unique elements in a list.
+
+    Should be in Python.
+    """
+    return [item for item, count in Counter(l).items() if count > 1]
+
+
+def validate_configuration(config):
+    messages = []
+    if not config.get('EMARSYS_ACCOUNT'):
+        messages.append(Critical("EMARSYS_ACCOUNT not set"))
+
+    if not config.get('EMARSYS_PASSWORD'):
+        messages.append(Critical("EMARSYS_PASSWORD not set"))
+
+    if not config.get('EMARSYS_BASE_URI'):
+        messages.append(Critical("EMARSYS_BASE_URI not set"))
+
+    events = config.get('EMARSYS_EVENTS')
+    if not events:
+        events = {}
+        messages.append(Critical("EMARSYS_EVENTS not set"))
+
+    messages += flatten([_validate_event(event, params)
+                         for event, params
+                         in events.items()])
+
+    # logging can happen here if desired
+
+    return messages
+
+
+def _validate_event(event, params):
+    if not event:
+        return [Warning("invalid event name: '{event}'".format(event=event))]
+
+    messages = []
+
+    duplicate_names = duplicates([param[0] for param in params.values()])
+    if duplicate_names:
+        messages.append(
+            Warning("reused parameter name for event '{event}': '{keys}'"
+                    .format(keys=", ".join(duplicate_names), event=event)))
+
+    return messages + flatten(
+        _validate_event_param(event, arg, param)
+        for arg, param in params.items())
+
+
+def _validate_event_param(event, arg, param):
+    messages = []
+
+    if not _VALID_ARG.match(arg):
+        messages.append(
+            Error("invalid parameter arg for event '{event}': '{value}'"
+                  .format(value=arg, event=event)))
+
+    if not isinstance(param, tuple) or len(param) != 2:
+        messages.append(
+            Error("invalid parameter definition '{event}': '{arg}' => {param}"
+                  .format(param=param, arg=arg, event=event)))
```

Comment:
If you want to check for duplicate names, that needs to happen on the level of the event. Ergo on that level you need to know about params structure.

You can write the comprehension like this:
```python
return messages + flatten(
    _validate_event_param(event, arg, name, model)
    for arg, (name, model) in params.items())
```

But of course that still leaves us with `param[0]` above that and destructuring the tuple outside the comprehension probably complicates things. Hrm. Maybe a namedtuple is not so bad after all? :)